posts/espp-in-this-economy/index.qmd

```diff
@@ -0,0 +1,215 @@
+```
+
+## Lookback
+
+The best ESPPs offer a lookback feature.
+
+With a lookback, the discount is applied to the price at the start of an offering period or the price on the purchase date, _whichever is lower_.
+
+"Whichever is lower" makes it difficult to lose.
+
+When the stock rises in value, shares are purchased at a discount from the earlier price, leading to even higher built-in profit.
+
+When the stock falls in value, shares are purchased at a discount from the current market price on the purchase date.
+
+There is profit in either scenario as long as the shares are sold immediately to lock it in.
+
+To calculate ESPP profit for your own plan that has a lookback feature, try the tool on the right.
+
+Try different scenarios, including ones where the stock loses value over the purchase period.
+
+```{ojs}
+//| panel: sidebar
+
+html `<figcaption>ESPP Interest Rate Calculator</figcaption>`;
+
+viewof lookback_purchase_period_in_months = Inputs.select([3, 6, 12], {
+  label: "Purchase Period",
+  format: (m) => `${m} months`,
+  value: 6,
+});
+
+viewof lookback_discount_percent = Inputs.number([0, Infinity], {
+  label: "Discount (%)",
+  step: 1,
+  value: 15,
+});
+
+viewof lookback_price_at_start = Inputs.number({
+  label: "Stock Price (start)",
+  type: "number",
+  step: 0.5,
+  value: 100,
+});
+
+viewof lookback_price_at_purchase = Inputs.number([0, Infinity], {
+  label: "Stock Price (purchase date)",
+  type: "number",
+  step: 0.5,
+  value: 105,
+});
+
+
+html `<hr>`;
+
+{
+  if (lookback_price_at_start < lookback_price_at_purchase) {
+    return html `The stock <strong>rose</strong> in value.<hr>The ${lookback_discount_percent}% discount is applied to the <strong>Stock Price (start)</strong> price.`;
+  } else {
+    return html `The stock <strong>fell</strong> in value.<hr>The ${lookback_discount_percent}% discount is applied to the <strong>Stock Price (purchase date)</strong> price.`;
+  }
+}
+
+lookback_purchase_price_with_discount = (1.0 - (lookback_discount_percent / 100.0)) * Math.min(lookback_price_at_start, lookback_price_at_purchase);
+html `<hr>`;
+html `Each share is purchased for <strong>$${lookback_purchase_price_with_discount.toFixed(2)}</strong> and is immediately worth <strong>$${lookback_price_at_purchase.toFixed(2)}</strong>`;
+
+lookback_average_escrow_time_in_months = lookback_purchase_period_in_months / 2.0;
+lookback_average_escrow_time_in_years = lookback_average_escrow_time_in_months / 12.0;
+lookback_interest_rate = Math.pow(lookback_price_at_purchase / lookback_purchase_price_with_discount, 1.0 / lookback_average_escrow_time_in_years) - 1.0;
+
+html `<hr>`;
+html `Annualized interest rate: <strong class="fs-1">${(100.0 * lookback_interest_rate).toFixed(2)}%</strong>`;
+```
+
+## Caveats
+
+ESPP administration and tax law have caveats that change the math in some scenarios. However, I have never seen these issues alone make a plan undesirable.
```

Comment:
Are the items listed below the caveats? If so, it might make sense to make that a little more clear by putting these in a bulleted list.